### DISCUSSION TITLE
outside_bbox is working properly?

### DISCUSSION THREAD

**Discussion by jicastillow** [jsvine/pdfplumber#1163] [Ask for help with specific PDFs]
Hello,

I'm trying to use Page.outside_bbox to get the original PDF without the tables detected. I do something like this:

```
import pdfplumber
import pdfplumber.table

pdf = pdfplumber.open("contracts/Palasiet.pdf")
page_number = 0
page = pdf.pages[page_number]
im = page.to_image(resolution=75)
tables = page.find_tables()
im.draw_rects(map(lambda table: table.bbox, tables))
```

Showing the detected table in the PDF:

![image](https://github.com/jsvine/pdfplumber/assets/22378148/9a7539e9-fdb3-4bc3-9a08-54a69c9986ec)

However then I try to use Page.outside_bbox to exclude the objects inside the table BBox,

`page.outside_bbox(tables[0].bbox).to_image()`

but none is excluded, aparently:

![image](https://github.com/jsvine/pdfplumber/assets/22378148/8c71b2f2-c593-4a1c-a170-6978fe9a9b55)

Can you please help me to know if I'm doing something wrong?

The idea is to get the same document but without the table on it.

**Comment by jsvine** [comment]
Hi @jicastillow, you're not doing anything wrong at all. The method `.outside_bbox(...)` returns a `DerivedPage` object where the elements inside of that bbox are removed *from `pdfplumber`'s internal view*. But because it does not alter the page's overall (outer) bounding box does not, it does not affect the output of `.to_image(...)`.

For your goal, the first block of code you have should work, with one tweak:

```python
im.draw_rects(map(lambda table: table.bbox, tables), fill="white")
```

... replacing `"white"` with whatever color name or RGB triplet you prefer. You can also change the stroke via `stroke=...`.

**Comment by jicastillow** [comment]
Hi @jsvine, thanks for your quick response, worked perfectly!
Keep up the good work, very useful library.
